src/ui/UI.cpp

```diff
@@ -61,6 +61,11 @@ void UI::Process()
             UiEventQueue::Event e = eventQueue_->GetAndRemoveNextEvent();
             if(e.type != UiEventQueue::Event::EventType::invalid)
                 ProcessEvent(e);
+            
+            for(int32_t i = pages_.GetNumElements() - 1; i >= 0; i--)
```

Comment:
This loop needs to sit inside the `if(e.type != UiEventQueue::Event::EventType::invalid)` statement. An Invalid event returned from the queue means: there was no event to pull - so consequently there also was no user activity.

Reply:
Updated.